File: src/9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        s1 = str(x)
        return s1 == s1[::-1]
# @lc code=end

if __name__ == "__main__":
    s = Solution()
    print(s.isPalindrome(123))
    print(s.isPalindrome(-121))
    print(s.isPalindrome(121))
    print(s.isPalindrome(0))

File: src/9.回文数.go
package main

/*
 * @lc app=leetcode.cn id=9 lang=golang
 *
 * [9] 回文数
 */

// @lc code=start
func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	f := func(x int) int {
		if x == 0 {
			return 0
		}
		flag, r := 1, 0
		if x < 0 {
			x, flag = -x, -1
		}
		for ; x%10 == 0; x /= 10 {
		}
		for ; x != 0; x /= 10 {
			r = r*10 + x%10
		}
		r *= flag
		return r
	}
	return x == f(x)
}

// @lc code=end
// func main() {
// 	fmt.Print(isPalindrome(-123))
// }
